pkg/cache/listers.go

```diff
@@ -0,0 +1,148 @@
+package cache
+
+import (
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/tools/cache"
+)
+
+// GenericLister is a lister skin on a generic Indexer
+type GenericClusterLister interface {
+	// List will return all objects across clusters
+	List(selector labels.Selector) (ret []runtime.Object, err error)
+	// ByCluster will give you a GenericLister for one namespace
+	ByCluster(cluster logicalcluster.LogicalCluster) cache.GenericLister
+}
+
+// NewGenericClusterLister creates a new instance for the genericClusterLister.
+func NewGenericClusterLister(indexer cache.Indexer, resource schema.GroupResource) *genericClusterLister {
+	return &genericClusterLister{indexer: indexer, resource: resource}
+}
+
+type genericClusterLister struct {
+	indexer  cache.Indexer
+	resource schema.GroupResource
+}
+
+func (s *genericClusterLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
+	if selector == nil {
+		selector = labels.NewSelector()
+	}
+	err = cache.ListAll(s.indexer, selector, func(m interface{}) {
+		ret = append(ret, m.(runtime.Object))
+	})
+	return ret, err
+}
+
+func (s *genericClusterLister) ByCluster(cluster logicalcluster.LogicalCluster) cache.GenericLister {
+	return &genericLister{indexer: s.indexer, resource: s.resource, cluster: cluster}
+}
+
+type genericLister struct {
+	indexer  cache.Indexer
+	cluster  logicalcluster.LogicalCluster
+	resource schema.GroupResource
+}
+
+func (s *genericLister) List(selector labels.Selector) (ret []runtime.Object, err error) {
+	selectAll := selector == nil || selector.Empty()
+	list, err := s.indexer.ByIndex(ClusterIndexName, s.cluster.String())
+	if err != nil {
+		return nil, err
+	}
+
+	if selector == nil {
+		selector = labels.Everything()
+	}
```

Comment:
Do you need this given the selectAll check above?

Reply:
sure don't